lib/rubocop/runner.rb

```diff
@@ -64,6 +64,10 @@ def aborting?
     # instances that each inspects its allotted group of files.
     def warm_cache(target_files)
       saved_options = @options.dup
+      if target_files.length <= 1
```

Comment:
Btw, shouldn't this be the first line of the method?

Reply:
If we make this the first line of the method then the `ensure` clause needs to have a separate `begin` block, otherwise it will replace `@options` with nil. I opted for this way because it looked slightly cleaner to my eyes and needed less changes.

Would you like me to rewrite it to skip the `.dup`?

Reply:
I see. I didn't notice the `ensure`. Probably it would have been nicer to have here `if/else` instead of `if` + `return`, although I assume RuboCop forced you to write the code like this.

Reply:
The following passes tests and rubocop inspection:

```
    # Warms up the RuboCop cache by forking a suitable number of RuboCop
    # instances that each inspects its allotted group of files.
    def warm_cache(target_files)
      if target_files.length <= 1
        puts 'Skipping parallel inspection: only a single file needs inspection' if @options[:debug]
      else
        begin
          saved_options = @options.dup
          puts 'Running parallel inspection' if @options[:debug]
          %i[autocorrect safe_autocorrect].each { |opt| @options[opt] = false }
          Parallel.each(target_files) { |target_file| file_offenses(target_file) }
        ensure
          @options = saved_options
        end
      end
    end
```